roles/addons/diskless/files/disklessset.py

```diff
@@ -539,17 +540,14 @@ def generate_ipxe_boot_file(image_type, image_name, image_kernel, image_initramf
             exit(1)
 
         print(bcolors.OKBLUE+'[INFO] Updating image files.'+bcolors.ENDC)
-        file = open(os.path.join(images_path, selected_image_name, 'boot.ipxe'), 'r')
-        filebuffer = file.readlines()
-        for i in range(len(filebuffer)):
-            if 'image-kernel' in filebuffer[i]:
-                filebuffer[i] = 'set image-kernel '+kernel_list[selected_kernel]+'\n'
-            if 'image-initramfs' in filebuffer[i]:
-                filebuffer[i] = 'set image-initramfs '+'initramfs-kernel-'+kernel_list[selected_kernel].strip('vmlinuz-')+'\n'
-        file.close
-        file = open(os.path.join(images_path, selected_image_name, 'boot.ipxe'), 'w')
-        file.writelines(filebuffer)
-        file.close
+        with open(os.path.join(images_path, selected_image_name, 'boot.ipxe'), "r+") as ff:
+            filebuffer = ff.read()
+            filebuffer = re.sub('^set image-kernel .+', 'set image-kernel ' + kernel_list[selected_kernel], filebuffer, flags=re.MULTILINE)
+            filebuffer = re.sub('^set image-initramfs .+', 'set image-initramfs initramfs-kernel-' + kernel_list[selected_kernel].strip('vmlinuz-'), filebuffer, flags=re.MULTILINE)
+            ff.seek(0)
```

Comment:
I do not really get why you need seek. Can you explain to me ?

Reply:
The object is at the end of the file because of the read() line 544. We need to move the object at the beginning of the file before writing the filebuffer.

Reply:
Ok thanks for explanation 🙂